tmt/steps/report/junit.py

```diff
@@ -114,34 +336,49 @@ class ReportJUnitData(tmt.steps.report.ReportStepData):
 @tmt.steps.provides_method('junit')
 class ReportJUnit(tmt.steps.report.ReportPlugin[ReportJUnitData]):
     """
-    Save test results in JUnit format.
+    Save test results in chosen JUnit flavor format. When flavor is set to
+    custom, the ``template-path`` with a path to a custom template must be
+    provided.
 
     When ``file`` is not specified, output is written into a file
     named ``junit.xml`` located in the current workdir.
     """
 
     _data_class = ReportJUnitData
 
+    def check_options(self) -> None:
+        """ Check the module options """
+
+        if self.data.flavor == 'custom' and not self.data.template_path:
+            raise tmt.utils.ReportError(
+                "The 'custom' flavor requires the '--template-path' argument.")
+
+        if self.data.flavor != 'custom' and self.data.template_path:
+            raise tmt.utils.ReportError(
+                "The '--template-path' can be used only with '--flavor=custom'.")
+
     def prune(self, logger: tmt.log.Logger) -> None:
         """ Do not prune generated junit report """
 
     def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
         """ Read executed tests and write junit """
         super().go(logger=logger)
 
-        junit_xml = import_junit_xml()
-        suite = make_junit_xml(self)
+        self.check_options()
 
         assert self.workdir is not None
         f_path = self.data.file or self.workdir / DEFAULT_NAME
+
+        xml_data = make_junit_xml(
+            phase=self,
+            flavor=self.data.flavor,
+            template_path=self.data.template_path,
+            include_output_log=self.data.include_output_log,
+            prettify=self.data.prettify)
         try:
             with open(f_path, 'w') as fw:
-                if hasattr(junit_xml, 'to_xml_report_file'):
-                    junit_xml.to_xml_report_file(fw, [suite])
-                else:
-                    # For older junit-xml
-                    junit_xml.TestSuite.to_file(fw, [suite])
-            self.info("output", f_path, 'yellow')
+                fw.write(xml_data)
+
+            self.info('output', f_path, 'yellow')
         except Exception as error:
-            raise tmt.utils.ReportError(
-                f"Failed to write the output '{f_path}' ({error}).")
+            raise tmt.utils.ReportError(f"Failed to write the output '{f_path}' ({error}).")
```

Comment:
Since you're changing this line, let's stop encoding `error` into message, and use proper `raise ... from error`.